src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js

```diff
@@ -58,7 +58,7 @@
          * @memberof BaseFieldValidator
          */
         toggleInvalidState(isError, config, input) {
-            const fieldNode = input.closest(this.fieldSelector);
+            const fieldNode = typeof this.fieldSelector === 'string' ? input.closest(this.fieldSelector) : this.fieldSelector;
```

Comment:
It allows to select a specific container instead of allowing to select a container by a query.
It's especially useful when you collect multiple elements in an array and you would like to apply the field preview functionality to these elements.

Reply:
This approach improves the performance of the JS code as well.